packages/rum-core/src/domain/tracing/tracer.ts

```diff
@@ -19,6 +19,22 @@ import type {
 import type { RumSessionManager } from '../rumSessionManager'
 import type { PropagatorType, TracingOption } from './tracer.types'
 
+/*
+Allows to control the behavior of trace context injection.
+This allows upstream services to inject trace context based on their sampling configuration.
+*/
+export const enum TraceContextInjection {
+  /*
+    Default. Inject trace context to all network requests as per the sampling rate.
+    */
+  All,
+
+  /*
+    Inject trace context only if the trace is sampled.
+    */
+  Sampled,
+}
```

Comment:
**💬 suggestion:** ‏user-facing enums should be defined as a plain object, like so:

https://github.com/DataDog/browser-sdk/blob/c52216b192f772911a07d0c8dd255a4134a5597b/packages/core/src/domain/configuration/configuration.ts#L17-L22

This allows using either the object (ex: `TraceContextInjection.All`) or the value directly (ex: `"all"`) without facing typing issues, from JS or TS.

In any case, `const enum` should never be customer-facing because non-TS users won't have access to it.